packages/genesys-spark-components/src/components/stable/gux-listbox/option-group/gux-option-group.tsx

```diff
@@ -9,17 +12,103 @@ import { Component, h, Host, JSX, Prop } from '@stencil/core';
   shadow: true
 })
 export class GuxOptionGroup {
+  private rootParent: HTMLElement;
+  private parentObserver: MutationObserver;
+
+  @Element() root: HTMLGuxOptionGroupBetaElement;
+
   @Prop()
   label!: string;
 
   // @Prop()
   // disabled: boolean;
   // TOOD: Add disabled state once design work is complete
 
+  @State()
+  private parentFilterValue: string = '';
+
+  @State()
+  filtered: boolean = false;
+
+  @State()
+  hideDivider: boolean = false;
+
+  componentDidLoad(): void {
+    this.rootParent = getClosestElement(
+      'gux-dropdown',
+      this.root
+    ) as HTMLElement;
+
+    this.parentObserver = onMutation(this.rootParent, () => {
+      this.hideDivider =
+        this.parentFilterValue.length > 0 &&
+        !this.hasVisibleGroupAfter(this.root);
+    });
+
+    if (this.rootParent) {
+      this.rootParent.addEventListener(
+        'guxfilter',
+        this.handleFilterChange.bind(this) as EventListenerOrEventListenerObject
+      );
+    }
+  }
+
+  disconnectedCallback(): void {
+    if (this.rootParent) {
+      this.rootParent.removeEventListener(
+        'guxfilter',
+        this.handleFilterChange.bind(this) as EventListenerOrEventListenerObject
+      );
+      this.parentObserver.disconnect();
+    }
+  }
+
+  private isOption(item: Element): boolean {
+    const optionTypes = ['GUX-OPTION', 'GUX-OPTION-ICON'];
+    return optionTypes.includes(item.tagName);
+  }
+
+  handleFilterChange(event: CustomEvent<string>) {
+    this.parentFilterValue = event.detail;
+
+    const options = Array.from(this.root?.children).filter(child =>
+      this.isOption(child)
+    ) as ListboxOptionElement[];
+    const allOptionsFiltered = !options.some(option =>
+      this.matchOption(option, this.parentFilterValue)
+    );
+
+    this.filtered = this.parentFilterValue.length > 0 && allOptionsFiltered;
+  }
+
+  private hasVisibleGroupAfter(element: Element): boolean {
+    let nextOption = element.nextElementSibling;
+    if (nextOption === null) return false;
+    while (nextOption && nextOption.classList.contains('gux-filtered')) {
+      nextOption = nextOption.nextElementSibling;
+    }
+    return Boolean(nextOption);
+  }
+
+  private matchOption(
+    option: ListboxOptionElement,
+    matchString: string
+  ): boolean {
+    return option.textContent
+      .trim()
+      .toLowerCase()
+      .startsWith(matchString.toLowerCase());
+  }
+
   render(): JSX.Element {
     return (
-      <Host>
-        <div class="gux-option-group">
+      <Host class={{ 'gux-filtered': this.filtered }}>
+        <div
+          class={{
+            'gux-option-group': true,
+            'gux-hide-divider': this.hideDivider
```

Comment:
I would  "hide" the divider by not rendering it if it is not necessary.
Something like
```
renderDivider(): JSX.Element {
  if (this.showDivider) {
    return ( <gux-list-divider></gux-list-divider> ) as JSX.Element;
  }
}

Reply:
Done